telegram/_message.py

```diff
@@ -1539,10 +1547,113 @@ def _quote(self, quote: Optional[bool], reply_to_message_id: Optional[int]) -> O
             else:
                 default_quote = None
             if (default_quote is None and self.chat.type != Chat.PRIVATE) or default_quote:
-                return self.message_id
+                return ReplyParameters(self.message_id)
 
         return None
 
+    def compute_quote_position_and_entities(
+        self, quote: str, index: Optional[int] = None
+    ) -> Tuple[int, Tuple[MessageEntity, ...]]:
+        if not (text := (self.text or self.caption)):
+            raise RuntimeError("This message has neither text nor caption.")
+
+        effective_index = 0 or index
+
+        matches = list(re.finditer(re.escape(quote), text))
+        if (length := len(matches)) < effective_index:
+            raise ValueError(
+                f"You requested the {index}-nth occurrence of '{quote}', but this text appears "
+                f"only {length} times."
+            )
+        # apply utf-16 conversion
+        position = matches[index].start()
+        end_position = position + len(quote)
+        entities = tuple(
+            entity
+            for entity in self.entities or self.caption_entities
+            if position <= entity.offset <= end_position
+        )
+        return position, entities
+
+    def build_reply_arguments(
+        self,
+        quote: Optional[str] = None,
+        quote_index: Optional[int] = None,
+        target_chat_id: Optional[Union[int, str]] = None,
+        allow_sending_without_reply: ODVInput[bool] = DEFAULT_NONE,
+        message_thread_id: Optional[int] = None,
+    ) -> _ReplyKwargs:
+        """
+        Use this function to build ``reply_paramters`` for ``telegram.Message.reply_*`` methods.
+
+            .. versionadded:: NEXT.VERSION
+
+        Args:
+            quote (:obj:`str`, optional): Quoted part of the message to be replied to; 0-1024
+                characters after entities parsing. The quote must be an exact substring of the
+                message to be replied to, including bold, italic, underline, strikethrough,
+                spoiler, and custom_emoji entities. The message will fail to send if the quote
+                isn't found in the original message.
+            quote_index (:obj:`int`, optional): Position of the quote in the original message in
+                UTF-16 code units.
+            target_chat_id (:obj:`int` | :obj:`str`, optional): If the message to be replied to is
+                from a different chat, |chat_id_channel|
+            allow_sending_without_reply (:obj:`bool`, optional): |allow_sending_without_reply|
+            message_thread_id (:obj:`int`, optional): |message_thread_id|
+
+        Returns:
+            :obj:`dict`: On success, a dict containing information about ``reply_parameters`` is
+            returned.
+        """
+        kwargs: _ReplyKwargs = {
+            "chat_id": target_chat_id or self.chat_id
+        }  # type: ignore [typeddict-item]
```

Comment:
if it helps with mypy, you can also construct `kwargs` at one place at the end of the method